File: 7.py
# Reverse Integer
x = -2147483648


def rev(x):
    l1 = []
    b = x
    while x > 0:
        rem = x % 10
        x //= 10
        l1.append(rem)

    length = len(l1)-1
    sum = 0
    for digit in l1:
        num = digit*(10**(length))
        sum += num
        (length) -= 1
    return sum


if x >= 0:
    out = rev(x)
    if out > 2147483647:
        print(0)
    else:
        print(out)
else:
    out = rev(-x)
    if out > 2147483648:
        print(0)
    else:
        print(-out)

File: 7.cpp
// Reverse Integer

#include <iostream>
using namespace std;

int main()
{
    int n;
    cout << "enter the number: ";
    cin >> n;

    int ans = 0;

    while (n != 0)
    {
        int digit = n % 10;
        if ((ans > INT_MAX / 10) || (ans < INT_MIN / 10))
        {
            cout << 0 << endl;
            break;
        }
        ans = (ans) * 10 + digit;
        n /= 10;
    }
    cout << "Reversed Integer: " << ans << endl;
}